docs/src/developers/testing.md

```diff
@@ -6,6 +6,174 @@ In order to run the unit tests run the following command
 ```bash
 make test
 ```
+### Adding tests
+General unit tests of functions follow the same conventions for testing using Go's `testing` standard library, along with the [testify](https://github.com/stretchr/testify) toolkit for making assertions.
+
+Unit tests that require API clients use mock clients generated using [gomock](https://github.com/uber-go/mock). To simplify the usage of mock clients, this repo also uses an internal library defined in `mock/mocktest`.
+
+`mocktest` is usually imported as a dot import along with the `mock` package:
+
+```go
+import (
+  "github.com/linode/cluster-api-provider-linode/mock"
+
+  . "github.com/linode/cluster-api-provider-linode/mock/mocktest"
+)
+```
+
+Using `mocktest` involves creating a test suite that specifies the mock clients to be used within each test scope and running the test suite using a DSL for defnining test nodes belong to one or more `Paths`.
+
+#### Example
+The following is a contrived example using the mock Linode machine client.
+
+Let's say we've written an idempotent function `EnsureInstanceRuns` that 1) gets an instance or creates it if it doesn't exist, 2) boots the instance if it's offline. Testing this function would mean we'd need to write test cases for all permutations, i.e.
+* instance exists and is not offline
+* instance exists but is offline, and is able to boot
+* instance exists but is offline, and is not able to boot
+* instance does not exist, and is not able to be created
+* instance does not exist, and is able to be created, and is able to boot
+* instance does not exist, and is able to be created, and is not able to boot
+
+While writing test cases for each scenario, we'd likely find a lot of overlap between each. `mocktest` provides a DSL for defining each unique test case without needing to spell out all required mock client calls for each case. Here's how we could test `EnsureInstanceRuns` using `mocktest`:
+
+```go
+func TestEnsureInstanceNotOffline(t *testing.T) {
+  suite := NewSuite(t, mock.MockLinodeMachineClient{})
+  
+  suite.Run(Paths(
```

Comment:
`Test` might be overloaded here, but what are your thoughts on changing `Paths` to `Tests` and `Path` to `Test`?

Reply:
That sounds good to me! I'll update it.